File: TwoPointer/791.custom-sort-string.py
# @leet start
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        freq = [0] * 128
        res = ""
        for c in s:
            freq[ord(c)] += 1
        for c in order:
            res += c * freq[ord(c)]
            freq[ord(c)] = 0
        for i, count in enumerate(freq):
            res += chr(i) * count
        return res


# @leet end

File: TwoPointer/791.custom-sort-string.go
package leetcode

import "strings"

// @leet start
func customSortString(order string, s string) string {
	var (
		res  strings.Builder
		freq = [128]int{}
	)
	for _, r := range s {
		freq[r]++
	}
	for _, r := range order {
		for range freq[r] {
			res.WriteRune(r)
		}
		freq[r] = 0
	}
	for i, count := range freq {
		for range count {
			res.WriteRune(rune(i))
		}
	}
	return res.String()
}

// @leet end

